bindings/python/dlite-python.i

```diff
@@ -969,14 +973,14 @@ int dlite_swig_set_scalar(void *ptr, DLiteType type, size_t size, obj_t *obj)
         }
 
         /* Assign id if not already provided */
-        if (n < 4)
-          s[3] = triple_get_id(NULL, s[0], s[1], s[2]);
+        if (n < 5)
+          s[4] = triple_get_id(NULL, s[0], s[1], s[2], s[3]);
```

Comment:
isn't n<5 also if datatype is not defined, while id is?

Reply:
I think this should work. The datatype has to be provided (possible as None) if id is provided, since a sequence with 5 elements must have a 4th element.